File: data/EXP01/golang_no_system_var/2564/2564.output_1.go
	"strconv"
	"strings"
)

func substringXorQueries(s string, queries [][]int) [][]int {
	n := len(s) // length of binary string s
	ans := make([][]int, len(queries))

	for i, q := range queries {
		first := strconv.Itoa(q[0])   // convert first integer to string
		second := strconv.Itoa(q[1])  // convert second integer to string
		left, right := -1, -1  // initialize left and right indices of substring
		
		for j := 0; j < n; j++ {
			for k := j; k < n; k++ {
				substr := s[j : k+1]  // extract substring from index j to k
				val, _ := strconv.ParseInt(substr, 2, 64)  // convert substring to decimal value
				
				if int(val) ^ q[0] == q[1] {
					if left == -1 || k-j < right-left {
						left = j
						right = k
					}
				}
			}
		}
		
		ans[i] = []int{left, right}
	}

	return ans
}

File: data/EXP01/python/2564/2564.output_1.py
# @lc app=leetcode id=2564 lang=python3
class Solution:
    def substringXorQueries(self, s: str, queries: List[List[int]]) -> List[List[int]]:
        n = len(s)
        prefix_xor = [0] * (n + 1)
        for i in range(n):
            prefix_xor[i + 1] = prefix_xor[i] ^ int(s[i])
        
        ans = []
        for query in queries:
            left, right = query
            xor_value = prefix_xor[left] ^ prefix_xor[right + 1]
            if xor_value == right - left:
                ans.append([left, right])
            else:
                ans.append([-1, -1])
        
        return ans